### PR TITLE
Fix hints list item rendering

### PR DESCRIPTION
**Pull request by nedredmond**
LC-1022

Removes legacy style overrides from `ul`

Accidentally fixes LC-1225 as well!

### CODE REVIEW

File: .changeset/tidy-radios-report.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@khanacademy/perseus": patch
+---
+
+CSS bug fix for hint renderer.
```
Comment (jeremywiebe) [inline .changeset/tidy-radios-report.md:5]: **suggestion**: Could you be more explicit here about what is fixed?  Something like "Fix list item rendering in HintRenderer (CSS bug fix)" or something.  I'm thinking into the future when we or someone else reviews the CHANGELOG and reads this line.

File: packages/perseus/src/styles/khan-exercise.css [hunk 2]
```diff
@@ -1,1492 +1,1496 @@
...
+        display: none;
+    }
+    p.question {
+        font-weight: bold;
+    }
+    var {
+        font-style: normal;
+    }
+
+    .hint.last-hint {
```
Comment (jeremywiebe) [inline packages/perseus/src/styles/khan-exercise.css:14]: **thought**: Related to above, could this be `.perseus-hint-renderer:last-child` instead?

File: packages/perseus/src/styles/perseus-renderer.less [hunk 3]
```diff
@@ -143,7 +143,7 @@
         margin: -11px 0px 22px 0px; // first-level lists need padding
     }
 
-    .paragraph ul:not(.perseus-widget-radio) {
+    div:not(.hint) .paragraph ul:not(.perseus-widget-radio) {
```
Comment (jeremywiebe) [inline packages/perseus/src/styles/perseus-renderer.less:146]: **suggestion**: Also related to above comments, can this just be `div:not(.perseus-hint-renderer)`?

**Review by jeremywiebe — COMMENTED** [review]
I had a few thoughts about maybe making less impactful changes to reduce the chance of regressions. Wrapping the `.framework-perseus` class around all of `khan-exercise.css` scares me a bit.  But my CSS knowledge isn't too strong. 

I'm going to leave this unapproved to give a chance for others on the Perseus team (or you, Ned) to chime in.  :)

File: packages/perseus/src/styles/khan-exercise.css (hunk 2 above)
Comment (nedredmond) [reply packages/perseus/src/styles/khan-exercise.css:14]: `perseus-hint-renderer` isn't added in some cases.

File: packages/perseus/src/styles/perseus-renderer.less (hunk 3 above)
Comment (nedredmond) [reply packages/perseus/src/styles/perseus-renderer.less:146]: Unfortunately, `perseus-hint-renderer` isn't added in some cases. I added `hint` to be added to the renderer in every case.

**Comment by nedredmond** [comment]
> Wrapping the `.framework-perseus` class around all of `khan-exercise.css` scares me a bit. But my CSS knowledge isn't too strong.

Don't we put `.framework-perseus` around every instance of Perseus?

**Comment by jeremywiebe** [comment]
> > Wrapping the `.framework-perseus` class around all of `khan-exercise.css` scares me a bit. But my CSS knowledge isn't too strong.
> 
> Don't we put `.framework-perseus` around every instance of Perseus?

We should, but only the `ArticleRenderer` does this within Perseus. All other instances, it's up to the hosting app to ensure that any Perseus renderer is nested inside a `.perseus-framework` element. :( I think we're probably safe doing this and in the future it'd be nice to have less finicky styles.

File: packages/perseus/src/styles/khan-exercise.css (hunk 2 above)
Comment (nedredmond) [reply packages/perseus/src/styles/khan-exercise.css]: @jeremywiebe also, concision is the opposite of what we need here. we need to evoke as many classes as possible for it to override global styles in webapp.

**Review by jeremywiebe — APPROVED** [review]

**Pull request merged**